corehq/apps/data_interfaces/templates/data_interfaces/partials/auto_update_rule_list.html

```diff
@@ -88,6 +104,11 @@ <h3>{% trans "Delete this rule?" %}</h3>
                 <span data-bind="text: action_error"></span>
               </p>
             </td>
+            <!-- ko if: $root.has_linked_data -->
+              <td data-bind="if: upstream_id">
```

Comment:
Above `upstream_id` is called (see [line 70](https://github.com/dimagi/commcare-hq/pull/33095/commits/83f47b74c6c15dbceeb768ed054600822cc158c6#diff-723938886de31a9b84f24b68cbb006898bd340bd7638445574738b340d7bbff8R70) for example), here it is not. Is that intentional or a mistake?

Reply:
Knockout still has some weird semantics I don't fully understand. I believe lines similar to 70 'call' `upstream_id` because of the conditional logic. See the bottom of [the knockout if binding docs](https://knockoutjs.com/documentation/if-binding.html) for an illustration. I can see an argument to make all implicit calls explicit, but I think knockout convention is to let observable expressions stand on their own.